react-native/react/login2/register/set-public-name/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+/* @flow */
+
+import React, {Component} from '../../../base-react'
+import {setDeviceName} from '../../../actions/login2'
+import Render from './index.render'
+
+export default class SetPublicName extends Component {
+  nameTaken (deviceName) {
+    return this.props.existingDevices && this.props.existingDevices.indexOf(deviceName) !== -1
```

Comment:
there is a story to plumb helper functions from go and i did go down that path a bit during the original kex2 but it turns out to be a lot of code. This data comes from the rpc and its just a simple filter so i'm not too worried. if we end up with a lot of business logic then its definitely worth doing